src/Core/src/PerformanceTracker/PerformanceProfiler.cs

```diff
@@ -0,0 +1,108 @@
+#nullable disable
+using System;
+using System.Diagnostics;
+
+namespace Microsoft.Maui.Performance;
+
+/// <summary>
+/// Provides performance tracking for various runtime operations such as layout passes, image loading, etc.
+/// </summary>
+internal static class PerformanceProfiler
+{
+	/// <summary>
+	/// Gets the layout performance tracker used to monitor layout-related operations.
+	/// </summary>
+	public static ILayoutPerformanceTracker Layout { get; private set; }
+
+	/// <summary>
+	/// Initializes the performance profiler with a performance tracker.
+	/// </summary>
+	/// <param name="layout">An instance of <see cref="ILayoutPerformanceTracker"/> to be used for tracking layout performance.</param>
+	/// <exception cref="ArgumentNullException">Thrown if <paramref name="layout"/> is null.</exception>
+	public static void Initialize(ILayoutPerformanceTracker layout)
+	{
+		Layout = layout ?? throw new ArgumentNullException(nameof(layout));
+	}
+
+	/// <summary>
+	/// Collects and returns performance statistics related to tracker operations.
+	/// </summary>
+	/// <returns>
+	/// A <see cref="PerformanceStats"/> object containing tracker metrics and the timestamp at which the data was collected.
+	/// </returns>
+	public static PerformanceStats GetStats()
+	{
+		if (Layout is null)
+			return PerformanceStats.Empty;
+
+		return new PerformanceStats
+		{
+			Layout = Layout.GetStats(),
+			TimestampUtc = DateTime.UtcNow
+		};
+	}
+
+	/// <summary>
+	/// Starts performance tracking for a given category and optional element.
+	/// </summary>
+	/// <param name="category">The category of operation being tracked (e.g. <see cref="PerformanceCategory.LayoutMeasure"/>).</param>
+	/// <param name="element">An optional identifier for the specific element being tracked (e.g. element name).</param>
+	/// <returns>A <see cref="PerformanceTracker"/> struct that must be explicitly stopped via <see cref="PerformanceTracker.Stop"/>.</returns>
+	public static PerformanceTracker Start(PerformanceCategory category, string element = null)
+	{
+		if (Layout is null)
+			return default;
+
+		return new PerformanceTracker(category, element);
+	}
+
+	/// <summary>
+	/// Represents a running performance tracking operation. Must call <see cref="Stop"/> to record the timing.
+	/// </summary>
+	internal readonly struct PerformanceTracker
```

Comment:
Maybe add a `IDisposable` here and add a `Dispose()` method. Then we can just do

```cs
using var performance = PerformanceProfiler.Start(PerformanceCategory.LayoutArrange, this.GetType().Name);
```

without having to worry about stopping.